### PR TITLE
VADC-904

### PR DESCRIPTION
**Pull request by tianj7**
https://ctds-planx.atlassian.net/browse/VADC-904

### New Features
Add username to workflow api

### CODE REVIEW

File: src/argowrapper/engine/helpers/argo_engine_helper.py [hunk 1]
```diff
@@ -51,9 +52,24 @@ def parse_common_details(
...
+
+    if user_name_str:
+        if user_name_str.startswith("user-"):
+            user_name_str = user_name_str[5:]
+        # argo engine encode % as -
+        user_name_str = urllib.parse.unquote(user_name_str.replace("-", "%"))
+        # put actual - back after decoding
+        user_name_str = user_name_str.replace("%", "-")
+
     return {
```
Comment (pieterlukasse) [inline src/argowrapper/engine/helpers/argo_engine_helper.py:69]: there is already a function `convert_username_label_to_gen3username` that does the conversion back. It is actually used and tested here: https://github.com/uc-cdis/argo-wrapper/blob/ae204e5658b4c451e51719b7b525fb48e8a7074c/test/test_argo_engine_helper.py#L22-L34

**Review by pieterlukasse — COMMENTED** [review]
I found an existing method in the code that already does what you added! Please take a look

File: src/argowrapper/engine/helpers/argo_engine_helper.py (hunk 1 above)
Comment (pieterlukasse) [reply src/argowrapper/engine/helpers/argo_engine_helper.py:69]: The whole thing is tested here: https://github.com/uc-cdis/argo-wrapper/blob/ae204e5658b4c451e51719b7b525fb48e8a7074c/test/test_argo_engine_helper.py#L50-L64

Comment (tianj7) [reply src/argowrapper/engine/helpers/argo_engine_helper.py]: moved functions from test file to helper class file.

File: src/argowrapper/engine/helpers/argo_engine_helper.py [hunk 2]
```diff
@@ -59,12 +58,7 @@ def parse_common_details(
         )
 
     if user_name_str:
-        if user_name_str.startswith("user-"):
-            user_name_str = user_name_str[5:]
-        # argo engine encode % as -
-        user_name_str = urllib.parse.unquote(user_name_str.replace("-", "%"))
-        # put actual - back after decoding
-        user_name_str = user_name_str.replace("%", "-")
+        user_name_str = convert_username_label_to_gen3username(user_name_str)
```
Comment (pieterlukasse) [inline src/argowrapper/engine/helpers/argo_engine_helper.py:61]: would be good to use this same `convert_username_label_to_gen3username` in https://github.com/uc-cdis/argo-wrapper/pull/134/commits/15b18656cdfb304bf5937ce65f5b627f4925e8e0#diff-c6a6579b39e448068e73736827d205c5b2b5b87829267f597e316b95f2f6a81cR85-R86

File: src/argowrapper/engine/helpers/argo_engine_helper.py [hunk 3]
```diff
@@ -51,9 +51,19 @@ def parse_common_details(
...
+            workflow_details["metadata"].get("labels").get(GEN3_USER_METADATA_LABEL)
+        )
+
+    if user_name_str:
+        user_name_str = convert_username_label_to_gen3username(user_name_str)
+
     return {
         "name": workflow_details["metadata"].get("name"),
         "phase": phase,
+        "userName": user_name_str,
```
Comment (pieterlukasse) [inline src/argowrapper/engine/helpers/argo_engine_helper.py:66]:
```suggestion
        "GEN3_USER_METADATA_LABEL": user_name_str,
```

**Review by pieterlukasse — COMMENTED** [review]
fix: use convert_username_label_to_gen3username in parse_details

**Review by pieterlukasse — DISMISSED** [review]

**Review by pieterlukasse — APPROVED** [review]

**Pull request merged**